### ISSUE TITLE
тестування і помилки

### ISSUE THREAD

**Issue by rokise** [pupenasan/PACFramework#53]
не працює посилання на файл drv.xcr
![image](https://user-images.githubusercontent.com/84643282/165967461-48385ef7-fd70-4941-9de3-db2a0137795a.png)

**Comment by Renorder** [comment]
Виявив парочку недоліків:
- На операторському екрані "VLVA" не спрацьовує команда блокування приводу. Також кнопки Відкрити/Закрити мабуть зайві, було б доцільніше кнопки ± 10.
- У навчальному прикладі, в розділі 5.2 невірно вказана [команда генерування файлів імпорту з мастерданних](https://github.com/pupenasan/PACFramework/blob/master/platforms/unitypro/deployex2.md#:~:text=node%20C%3A%5Cpacfwtools%5Cnode_modules%5Cpacframework-tools%5Cindex%20seuncreateacts).
А саме вказана команда: `node C:\pacfwtools\node_modules\pacframework-tools\index seuncreatevars`
Гадаю має бути команда: `node C:\pacfwtools\node_modules\pacframework-tools\index seuncreateall`
Ця помилка виникла після виправлення помилки розгортання.
- Відсутність у мастердаті каналів АО викликає помилку, при спробі сформувати файл "aovar.xcr".
Можливо подібне станеться при відсутності будь якого з інших типів. 
![image](https://user-images.githubusercontent.com/61793962/166652154-840f9d87-ddf7-4eec-8819-5014dd33d8ff.png)

**Comment by pupenasan** [comment]
> Відсутність у мастердаті каналів АО викликає помилку, при спробі сформувати файл "aovar.xcr".
Можливо подібне станеться при відсутності будь якого з інших типів.

Це у власному прикладі? Якщо так, то давайте свій excel мені, щоб я його протестив

**Comment by Renorder** [comment]
> Це у власному прикладі?

Так, це спостерігалось у власній мастердаті. Хоча, як щойно виявилося, те саме стосується і мастердати з прикладу, якщо видалити канали АО.
[masterdata.xlsx](https://github.com/pupenasan/PACFramework/files/8629268/masterdata.xlsx)

**Comment by pupenasan** [comment]
> > Це у власному прикладі?
> 
> Так, це спостерігалось у власній мастердаті. Хоча, як щойно виявилося, те саме стосується і мастердати з прикладу, якщо видалити канали АО. [masterdata.xlsx](https://github.com/pupenasan/PACFramework/files/8629268/masterdata.xlsx)

наче пофіксив. Треба обновити тулзу і знову попробувати

**Comment by Renorder** [comment]
> наче пофіксив. Треба обновити тулзу і знову попробувати

Оновив, все запрацювало)
- Команди генерування файлів імпорту нормально сприймаються.
- Відсутність каналів певного типу більше не викликає помилки.
- Генеруються всі необхідні файли.

Знайшов ще один не суттєвий нюанс. На етапі створення змінних та секцій ACTRS було попередження:
`WRN: Не вдалося завантажити файл C:\Users\Dibrov\pacframeworktools\source\drvs.xcr, перевірте наявність файлу`
Якщо я не помиляюся, назву вона шукає з колонки "FN" на сторінці "acttps" у мастердаті? Там є девайс `drvs`, а в [інструкції](https://github.com/pupenasan/PACFramework/blob/master/platforms/unitypro/deployex2.md#61-%D0%B7%D0%B0%D0%B2%D0%B0%D0%BD%D1%82%D0%B0%D0%B6%D0%B5%D0%BD%D0%BD%D1%8F-%D1%82%D0%B0-%D1%96%D0%BC%D0%BF%D0%BE%D1%80%D1%82%D1%83%D0%B2%D0%B0%D0%BD%D0%BD%D1%8F-%D0%B1%D0%B0%D0%B7%D0%BE%D0%B2%D0%B8%D1%85-%D1%82%D0%B8%D0%BF%D1%96%D0%B2-%D0%B2%D0%B8%D0%BA%D0%BE%D0%BD%D0%B0%D0%B2%D1%87%D0%B8%D1%85-%D0%BC%D0%B5%D1%85%D0%B0%D0%BD%D1%96%D0%B7%D0%BC%D1%96%D0%B2) є файл `drv.xcr`. Перейменувавши одне в друге я проблему вирішив. Просто в інструкції є така дрібна неточність.

Наразі більше проблем не бачу. Якщо ще щось знайду, то обов'язково сюди відпишу.

**Comment by pupenasan** [comment]
Файл `drv.xcr` треба перенести в source. Раніше його не було в репозиторії, зараз він уже є.
